frontend/src/app/login/login.component.ts

```diff
@@ -26,16 +30,17 @@ export class LoginComponent implements OnInit {
     submit(email: string, password: string): void {
         this.auth.login(email, password)
             .subscribe(
-            response => {
+                response => {
+                    this.userEmailService.setEmail(email); // set email for observer form submission
                     this.router.navigate(['dashboard']);
-            },
-            error => {
-                if (error.status === 401) {
-                    alert('login failed');
-                    if (this.loginAttempts-- === 0) {
-                        this.router.navigate(['password-recovery']);
+                },
+                error => {
+                    if (error.status === 401) {
+                        alert('login failed');
+                        if (this.loginAttempts-- === 0) {
```

Comment:
I'd prefer if this check was separated. It makes the code harder to read to have conditional checks and assignments on the same line.